src/commands/apphosting-backends-delete.ts

```diff
@@ -47,21 +45,27 @@ export const command = new Command("apphosting:backends:delete")
       });
     }
 
+    let backend: apphosting.Backend;
+    if (backendId) {
```

Comment:
Okay, this is odd. According to the approved proposal, the backend ID is supposed to be a required parameter, not a flag. That eliminates the need to pick a backend at all.

Reply:
Irrespective of the proposal, do you like this better? It seemed odd to me that location could be browsed but backend couldn't. If you like this better we can push it through API council. As an EAP API, do you feel like it is OK to release and then get approval, or should I splice this into another CL until after approval?

Reply:
I personally don't like it better, because I hate the prompt selector flow in a CLI 🤷‍♂️ It's not a wrong thing to do, but I'm assuming that there were product discussions that brought us to the API proposal (as that's a requirement at the top of the document) and that the pattern in the proposal was agreed upon by all the relevant stakeholders already. If you'd like to make a change to what was proposed there, starting back in the process, involving the right people, and getting an addendum would be appropriate.

That said, I thought the general pattern was to:

1. require backendID as a parameter in the command
2. default to looking at us-central1 as we 'traditionally' have

If someone is outside of us-central1, they get to include the `--location` flag.

Having this flow also makes it more complex. Sticking with the picker, you have to also consider what happens in non-interactive environment (like a CI system). As written, this would hang, I believe, which may be unexpected (instead of simply failing out).

Reply:
(If it's chosen that we should keep the existing dialog, I agree that we need to check for nonInteractive)